addons/product_expiry/models/production_lot.py

```diff
@@ -59,30 +59,45 @@ def _onchange_product(self):
     @api.model
     def _alert_date_exceeded(self):
         # if the alert_date is in the past and the lot is linked to an internal quant
-        # log a next activity on the next.production.lot 
+        # log a next activity on the next.production.lot
         alert_lot_ids = self.env['stock.production.lot'].search([('alert_date', '<=', fields.Date.today())])
-        mail_activity_type = self.env.ref('product_expiry.mail_activity_type_alert_date_reached').id
+        mail_activity_type = self.env.ref('product_expiry.mail_activity_type_alert_date_reached', raise_if_not_found=False)
+        if mail_activity_type:
```

Comment:
Where does this code modification comes from ?

Reply:
It is about the _onchange_activity_type_id() called just after. As I understood it, it is called to update activity deadline depending on the activity type, but it could maybe be removed now.

Reply:
Refactored in independent commit.

Reply:
Moved to https://github.com/odoo/odoo/pull/33300.